File: 9. Palindrome Number/isPalindrome.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        origin = x
        new = 0
        while x > 0:
            temp = x % 10
            new = new * 10 + temp
            x = (x - temp) // 10
        return origin == new 

File: 9. Palindrome Number/isPalindrome.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        String xStr = String.valueOf(x);
        int left = 0, right = xStr.length() - 1;
        while (left < right) {
            if (xStr.charAt(left) != xStr.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}